File: 문제풀이/프로그래머스_고득점_KIT/해시/의상.py
from collections import defaultdict
from math import prod

def solution(clothes):
    cloth_dict = defaultdict(list)
    for cloth, cloth_type in clothes:
        cloth_dict[cloth_type].append(cloth)
    
    answer = prod(len(clothes) + 1 for clothes in cloth_dict.values()) - 1
    return answer

'''
풀이 시간: 6분
접근:
후기:
    from collections import Counter을 이용해서 셀 수 있다.

    cnt = Counter([kind for name, kind in clothes])
'''

File: CPP/프로그래머스_고득점_KIT/해시/의상.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    map<string, int> cnt;
    
    for (auto& cloth : clothes) {
        cnt[cloth[1]]++;
    }
    
    int answer = 1;
    
    for (auto& pair : cnt) {
        answer *= pair.second + 1;
    }
    
    answer--;
    
    return answer;
}

/*
풀이 시간: 5분 20초
접근:
    옷의 종류별 개수를 센다.
    
    그리고 각 개수를 곱한 뒤 1을 빼준다. (옷을 안 입은 경우를 빼준다.)

후기:
    문법에 익숙해지는 중..

*/